### DISCUSSION TITLE
High frequency of V-gene chimeras during analysis of non-model organism

### DISCUSSION THREAD

**Discussion by gyorkeia** [milaboratory/mixcr#1136] [Q&A] [answered]
Hi,
I am working on characterizing the BCRH repertoire of the Atlantic cod. I have used the genome reference to create an exhaustive list of V/D/J/C genes, that I turned into a RepSeqIO library. However, when I map my reads to it (5'RACE sequencing, PE, 150x2 length), I get a very high frequency of chimeric V-genes:
V gene chimeras: 458788 (45.88%), from a trial run of 1M reads.
An important thing to note is that the reference genome I use is not the same family as the bred line from which the samples are from. The latter one has no reference genome and it is currently impossible to say what the heterogeneity in gene and allelic frequency is between these two. 
Could this be the attributed to the difference between the genetic background or is there anything else that may cause it. 
I have not found a clear description on what chimeras are based on what cutoffs are.
Is there any downstream results that are affected by this and can this issue be addressed somehow?

Thanks.
Adam

Ps:please find the whole report below:

Running:
mixcr align --report ./F0002_IgH_analysis.align.report.txt --json-report ./F0002_IgH_analysis.align.report.json --threads 16 --preset takara-mouse-bcr-full-length --limit-input 1000000 --rigid-left-alignment-boundary --floating-right-alignment-boundary C --rna --species gm --library IgH_all_final_renamed.json --tag-pattern ^N{0:6}gagtagcggaaagacagtgggggcttgca(
R1:*)\^N{0:6}caactaaagcagtggtatcaacgcagagt(R2:*) --dont-split-clones-by C --assemble-clonotypes-by CDR3 --dont-impute-germline-on-export ./F0002_1.fastq.gz ./F0002_2.fastq.gz ./F0002_IgH_analysis.vdjca
WARNING: 5 functional genes were excluded, re-run with --verbose option to see the list of excluded genes and exclusion reason.
Alignment: 0%
Alignment: 10.5%  ETA: 00:03:42
Alignment: 20.7%  ETA: 00:02:34
Alignment: 30.9%  ETA: 00:02:15
Alignment: 41.1%  ETA: 00:01:55
Alignment: 51.2%  ETA: 00:01:36
Alignment: 61.3%  ETA: 00:01:17
Alignment: 71.4%  ETA: 00:00:56
Alignment: 81.5%  ETA: 00:00:36
Alignment: 91.6%  ETA: 00:00:16
============== Align Report ==============
Analysis time: 3.39m
Total sequencing reads: 1000000
Successfully aligned reads: 885490 (88.55%)
Paired-end alignment conflicts eliminated: 136 (0.01%)
Alignment failed, no hits (not TCR/IG?): 1453 (0.15%)
Alignment failed because of absence of V hits: 6513 (0.65%)
Alignment failed because of absence of J hits: 1836 (0.18%)
No target with both V and J alignments: 7164 (0.72%)
Absent barcode: 97544 (9.75%)
Overlapped: 1154 (0.12%)
Overlapped and aligned: 352 (0.04%)
Alignment-aided overlaps: 11 (3.12%)
Overlapped and not aligned: 802 (0.08%)
No CDR3 parts alignments, percent of successfully aligned: 213 (0.02%)
Partial aligned reads, percent of successfully aligned: 3968 (0.45%)
V gene chimeras: 458788 (45.88%)
IGH chains: 885490 (100%)
IGH non-functional: 70808 (8%)
Realigned with forced non-floating bound: 1802626 (180.26%)
Realigned with forced non-floating right bound in left read: 29277 (2.93%)
Realigned with forced non-floating left bound in right read: 29277 (2.93%)
Trimming report:
  R1 Reads Trimmed Left: 180 (0.02%)
  R1 Reads Trimmed Right: 57 (0.01%)
  Average R1 Nucleotides Trimmed Left: 4.77E-4
  Average R1 Nucleotides Trimmed Right: 0.001439
  R2 Reads Trimmed Left: 733 (0.07%)
  R2 Reads Trimmed Right: 215 (0.02%)
  Average R2 Nucleotides Trimmed Left: 0.002371
  Average R2 Nucleotides Trimmed Right: 0.002916
Tag parsing report:
  Execution time: 0ns
  Total reads: 1000000
  Matched reads: 902456 (90.25%)
  Projection +R1 +R2: 902456 (90.25%)
  For variant 0:
    For projection [1, 2]:
      R1:Left position: 35
      R1:Right position: 151
      R2:Left position: 35
      R2:Right position: 151
      Variants: 0
      Cost:
        0: + 828045 (91.75%) = 828045 (91.75%)
        5~11: + 74411 (8.25%) = 902456 (100%)
      R1 length: 116
      R2 length: 116
====================
Running:
mixcr assemble --report ./F0002_IgH_analysis.assemble.report.txt --json-report ./F0002_IgH_analysis.assemble.report.json ./F0002_IgH_analysis.vdjca ./F0002_IgH_analysis.clns
Initialization: progress unknown
Assembling initial clonotypes: 20.1%
Assembling initial clonotypes: 46.6%  ETA: 00:00:02
Assembling initial clonotypes: 73.8%  ETA: 00:00:00
Assembling initial clonotypes: 99.7%  ETA: 00:00:00
Mapping low quality reads: 12.4%
Mapping low quality reads: 28.9%  ETA: 00:00:04
Mapping low quality reads: 47.3%  ETA: 00:00:02
Mapping low quality reads: 67.2%  ETA: 00:00:01
Mapping low quality reads: 86.3%  ETA: 00:00:00
Pre-clustering: progress unknown
Clustering: 0.2%
Clustering: 10.6%  ETA: 00:00:34
Clustering: 24.2%  ETA: 00:00:22
Clustering: 34.4%  ETA: 00:00:19
Clustering: 44.8%  ETA: 00:00:15
Clustering: 55.8%  ETA: 00:00:12
Clustering: 66.5%  ETA: 00:00:09
Clustering: 80.9%  ETA: 00:00:05
Clustering: 92.6%  ETA: 00:00:01
Building clones: 6.3%
Building clones: 20.1%  ETA: 00:00:11
Building clones: 33.2%  ETA: 00:00:10
Building clones: 46.1%  ETA: 00:00:08
Building clones: 58.8%  ETA: 00:00:06
Building clones: 71.5%  ETA: 00:00:04
Building clones: 84%  ETA: 00:00:02
Building clones: 96.3%  ETA: 00:00:00
============== Assemble Report ==============
Analysis time: 57.66s
Final clonotype count: 77972
Average number of reads per clonotype: 10.52
Reads used in clonotypes, percent of total: 820594 (82.06%)
Reads used in clonotypes before clustering, percent of total: 831493 (83.15%)
Number of reads used as a core, percent of used: 798713 (96.06%)
Mapped low quality reads, percent of used: 32780 (3.94%)
Reads clustered in PCR error correction, percent of used: 10899 (1.31%)
Reads pre-clustered due to the similar VJC-lists, percent of used: 0 (0%)
Reads dropped due to the lack of a clone sequence, percent of total: 4267 (0.43%)
Reads dropped due to a too short clonal sequence, percent of total: 11 (0%)
Reads dropped due to low quality, percent of total: 0 (0%)
Reads dropped due to failed mapping, percent of total: 49713 (4.97%)
Reads dropped with low quality clones, percent of total: 0 (0%)
Clonotypes eliminated by PCR error correction: 3078
Clonotypes dropped as low quality: 0
Clonotypes pre-clustered due to the similar VJC-lists: 0
Clones dropped in post filtering: 0 (0%)
Reads dropped in post filtering: 0.0 (0%)
Alignments filtered by tag prefix: 0 (0%)
IGH chains: 77972 (100%)
IGH non-functional: 14806 (18.99%)
====================
Running:
mixcr exportClones ./F0002_IgH_analysis.clns ./F0002_IgH_analysis.clones.tsv
Exporting IGH
Exporting clones: 0%
Exporting clones: 35.8%  ETA: 00:00:01
Analysis finished successfully.
Grouping by the same J gene: 0%
Grouping by the same J gene: 65.2%  ETA: 00:00:00
Searching for J alleles: 65.2%
Grouping by the same V gene: 65.2%
Recalculating scores F0002_IgH_analysis.clns: 10.5%
Realigning F0002_IgH_analysis.clns: 0.2%
Realigning F0002_IgH_analysis.clns: 11.8%  ETA: 00:00:07
Realigning F0002_IgH_analysis.clns: 69.1%  ETA: 00:00:01
Realigning F0002_IgH_analysis.clns: 100%  ETA: 00:00:00

**Comment by gyorkeia** [comment]
hello?

**Comment by Alex-Davydov** [comment, marked as answer]
Hi,
`V gene chimeras` in the alignment report indicates the occurrences where distinct V genes were matched in separate paired-end reads. A high frequency of such occurrences for V genes indicates strong evidence of potential issues with the sample preparation process. To ensure the expected library structure, it is recommended to inspect the alignments using the `mixcr exportAlignmentsPretty` command. This process will enable you to visually review the alignments and verify the correctness of the library preparation procedure.
